crypto/lakers-crypto-psa/src/lib.rs

```diff
@@ -55,19 +55,18 @@ impl CryptoTrait for Crypto {
         message[..info.len()].copy_from_slice(info);
         message[info.len()] = 0x01;
         let mut t_i = self.hmac_sha256(&message[..info.len() + 1], prk);
-        output[..SHA256_DIGEST_LEN].copy_from_slice(&t_i);
+        let t_i_len = core::cmp::min(result.len(), SHA256_DIGEST_LEN);
+        result[..t_i_len].copy_from_slice(&t_i[..t_i_len]);
 
         for i in 2..=n {
             message[..SHA256_DIGEST_LEN].copy_from_slice(&t_i);
             message[SHA256_DIGEST_LEN..SHA256_DIGEST_LEN + info.len()].copy_from_slice(&info);
             message[SHA256_DIGEST_LEN + info.len()] = i as u8;
             t_i = self.hmac_sha256(&message[..SHA256_DIGEST_LEN + info.len() + 1], prk);
-            output[(i - 1) * SHA256_DIGEST_LEN..i * SHA256_DIGEST_LEN].copy_from_slice(&t_i);
+            let start = (i - 1) * SHA256_DIGEST_LEN;
+            let t_i_len = core::cmp::min(result[start..].len(), SHA256_DIGEST_LEN);
+            result[start..start + t_i_len].copy_from_slice(&t_i[..t_i_len]);
```

Comment:
Could you remind me if we have explicit test vectors that test HKDF-expand and -extract? I suppose CI not complaining means that the changes are OK.

Reply:
I don't know if we have any; they are generally covered by the fuller EDHOC tests.

For this case specifically (the extraction output being longer than a single block), I'll do a run where the output gets deliberately broken to see if that case is covered too. (Usually we'd try to avoid that case, to the point where I've seen HKDF implementations that don't even support the more-than-one-block form).

Reply:
Local tests broke with

```diff
diff --git a/crypto/lakers-crypto-rustcrypto/src/lib.rs b/crypto/lakers-crypto-rustcrypto/src/lib.rs
index 93a7709..cabc828 100644
--- a/crypto/lakers-crypto-rustcrypto/src/lib.rs
+++ b/crypto/lakers-crypto-rustcrypto/src/lib.rs
@@ -54,2 +54,5 @@ impl<Rng: rand_core::RngCore + rand_core::CryptoRng> CryptoTrait for Crypto<Rng>
             .expect("Static lengths match the algorithm");
+        if result.len() > 32 {
+            result[result.len() - 1] ^= 0x55;
+        }
     }
```

in the test `tests::test_hkdf_expand::case_2` -- so yes, we do have a test case covering those.